### PR TITLE
Feat single plane

### PR DESCRIPTION
**Pull request by arielleleon**

### CODE REVIEW

File: code/registration.py [hunk 1]
```diff
@@ -1153,16 +1154,17 @@ def get_frame_rate_from_sync(sync_file, platform_data) -> float:
...
 
     Parameters
     ----------
-    datainput: Path
+    input: Path
         path to h5 file
     output_dir: Path
         output directory
-
+    unique_id: str
```
Comment (j-friedrich) [inline code/registration.py:1166]: there's no argument `unique_id`, but `debug`

File: code/registration.py [hunk 2]
```diff
@@ -1211,23 +1208,153 @@ def multiplane_motion_correction(datainput: Path, output_dir: Path, debug: bool
...
+        path to h5 file
+    session: dict
+        session metadata
+    Returns
+    -------
+    Path
+        path to reference image
+    """
+    with h5py.File(fp, "r") as f:
+        data = f["data"][:]
```
Comment (j-friedrich) [inline code/registration.py:1322]: This loads the entire movie into RAM.

File: code/registration.py [hunk 3]
```diff
@@ -1211,23 +1208,153 @@ def multiplane_motion_correction(datainput: Path, output_dir: Path, debug: bool
...
+        dtype = data.dtype
+        # take the first bci epoch to save out reference image TODO
+        tiff_stems = json.loads(f["tiff_stem_location"][:][0])
+        bci_epochs = [
+            i
+            for i in session["stimulus_epochs"]
+            if i["stimulus_name"] == "single neuron BCI conditioning"
+        ]
+        bci_epoch_loc = [i["output_parameters"]["tiff_stem"] for i in bci_epochs][0]
+        with h5py.File("../scratch/reference_image.h5", "w") as ref:
```
Comment (j-friedrich) [inline code/registration.py:1332]: Can't we just create a VDS? Isn't that a movie not a single image?

File: code/registration.py [hunk 4]
```diff
@@ -1211,23 +1208,153 @@ def multiplane_motion_correction(datainput: Path, output_dir: Path, debug: bool
...
     Parameters
     ----------
-    datainput: Path
+    h5_file: Path
         path to h5 file
     output_dir: Path
         output directory
+    unique_id: str
+        experiment id from data description
+    session: dict
```
Comment (j-friedrich) [inline code/registration.py:1356]: in the signature of the function `session` comes before `unique_id`

File: code/registration.py [hunk 5]
```diff
@@ -1236,51 +1363,32 @@ def singleplane_motion_correction(datainput: Path, output_dir: Path, debug: bool
...
-    output_h5_file = Path(output_dir) / "bergamo.h5"
+    if not h5_file.is_file():
+        h5_file = [f for f in h5_file.rglob("*.h5") if unique_id in str(f)][0]
+    print(f"Running h5 file: {h5_file}")
+    output_dir = make_output_directory(output_dir, unique_id)
+    reference_image_fp = generate_bergamo_movies(h5_file, session)
+    if debug:
+        stem = h5_file.stem
+        debug_file = Path("../scratch") / f"{stem}_debug.h5"
+        with h5py.File(h5_file, "r") as f:
```
Comment (j-friedrich) [inline code/registration.py:1377]: Fine, though 5000 seems still quite a lot of frames for debugging purposes, and we could have used a virtual dataset.

File: code/registration.py [hunk 6]
```diff
@@ -1440,28 +1548,41 @@ def singleplane_motion_correction(datainput: Path, output_dir: Path, debug: bool
     # Parse command-line arguments
     args = parser.parse_args()
     # General settings
-    datainput = Path(args.input_searchpath)
+    datainput = Path(args.input)
     output_dir = Path(args.output_dir)
     data_dir = Path("../data")
```
Comment (j-friedrich) [inline code/registration.py:1553]: Isn't `data_dir` the same as `datainput`?

File: code/registration.py [hunk 7]
```diff
@@ -1211,23 +1208,153 @@ def multiplane_motion_correction(datainput: Path, output_dir: Path, debug: bool
...
-def singleplane_motion_correction(datainput: Path, output_dir: Path, debug: bool = False):
+def update_suite2p_args_reference_image(
+    suite2p_args: dict, args: dict, reference_image_fp=None
+):
+    # Use our own version of compute_reference to create the initial
+    # reference image used by suite2p.
+    logger.info(
+        f'Loading {suite2p_args["nimg_init"]} frames ' "for reference image creation."
+    )
+    if reference_image:
```
Comment (j-friedrich) [inline code/registration.py:1227]: is that supposed to be `reference_image_fp` instead of `reference_image`?

File: code/registration.py (hunk 6 above)
Comment (arielleleon) [reply code/registration.py:1553]: I thought I fixed that - must have been a different branch

File: code/registration.py (hunk 5 above)
Comment (arielleleon) [reply code/registration.py:1379]: Adding this as an issue

File: code/registration.py (hunk 3 above)
Comment (arielleleon) [reply code/registration.py:1339]: This is done

**Pull request merged**